### PR TITLE
Add a simple blacklist

### LINKED ISSUE #29: False positive: name ALWAYS assumed to be SQL keyword
In many contexts, `name` is not actually a SQL keyword, but sqlup-mode always treats it as one.

Example:

    INSERT INTO roles(create_timestamp, name, description)

is incorrectly sqlup'ed to

    INSERT INTO roles(create_timestamp, NAME, description)

### PR DESCRIPTION
**Pull request by davidshepherd7**
Fixes #29

### CODE REVIEW

File: sqlup-mode.el [hunk 1]
```diff
@@ -59,6 +59,19 @@
...
+
+;;; Customizable variables
+(defcustom sqlup-blacklist
+  '()
+  "List of words which should never be upcased
+
+The words must match the whole symbol. They are interpreted as plain
+strings not regexes."
+  :type '(repeat string)
+  :group 'sqlup)
```
Comment (Trevoke) [inline sqlup-mode.el]: Ooh, this is beginning to feel like a real package :)

File: sqlup-mode.el [hunk 2]
```diff
@@ -210,8 +224,10 @@ ANSI SQL keywords."
   (and (boundp 'sql-mode-font-lock-keywords) sql-mode-font-lock-keywords))
 
 (defun sqlup-keyword-p (word)
-  (cl-some (lambda (reg) (string-match-p (concat "^" reg "$") word))
-           (sqlup-keywords-regexps)))
+  (and (cl-some (lambda (reg) (string-match-p (concat "^" reg "$") word))
+                (sqlup-keywords-regexps))
+       (not (cl-some (lambda (blacklisted) (string-match-p (concat "^" (regexp-quote blacklisted) "$") word))
```
Comment (Trevoke) [inline sqlup-mode.el]: I think this is separate from the question of whether this is an SQL keyword or not: it's a matter of whether or not we should upcase it. So I think the call to this (I dunno, `sqlup-excluded-keyword-p` maybe?) should probably be in `sqlup-capitablizable-p`.

Thoughts?

File: sqlup-mode.el (hunk 1 above)
Comment (davidshepherd7) [inline sqlup-mode.el]: :+1:

File: sqlup-mode.el (hunk 2 above)
Comment (davidshepherd7) [inline sqlup-mode.el]: Yeah, you're right, I've moved it now.

**Comment by Trevoke** [comment]
This is fantastic. The last bit is to write a little documentation for this (I usually just write it once and duplicate the text in the README and in the top comment block of `sqlup-mode.el`.

The documentation can be very sparse: we're in pre-1.0, still, and this is literally an experiment to see if it's good enough as a feature for package users.

Are you up for it? If not, I can do it.

**Comment by davidshepherd7** [comment]
Oops yeah, I was intending to add some documentation but then forgot, done now.

**Comment by Trevoke** [comment]
Awesome, thank you!

**Pull request merged**